### ISSUE TITLE
Feature suggestion: Use IPinfo's IP to Country ASN Database as the primary IP database source.

### ISSUE THREAD

**Issue by abdullahdevrel** [rennf93/fastapi-guard#7]
I am requesting to add support for [IPinfo's IP to Country database](https://ipinfo.io/products/free-ip-database) to the project. The database has the following features:

The current implementation of IPinfo's service involves calling the API. For a web service, this is not scalable. That is why we offer the free IP to Country ASN/ISP database. The database can handle an infinite number of queries and includes ASN/ISP metadata, adding a layer of IP metadata dimension. In my opinion, ASN is a more common metadata for blacklisting as it identifies different hosting services (such as AWS, Digital Ocean, Tencent, etc.) that can operate in multiple countries.

- It includes country and ASN information in the same database.
- It is updated daily, with zero compromise to accuracy. There is no range clustering, and the database provides full accuracy.
- The data granularity reaches individual IP level.
- The database comes in MMDB database format.
- It is licensed under CC-BY-SA 4.0, permitting commercial usage.
- Available file formats include: CSV, MMDB, JSON
- The data is tabular and unnested, making it very easy to use. The dataset includes both IPv4 and IPv6 in a single file.

## Database schema

| Field Name       | Example                      | Data Type | Description                                       |
|------------------|------------------------------|-----------|---------------------------------------------------|
| `start_ip`       | 1.0.16.0                     | TEXT      | Starting IP address of an IP address range        |
| `end_ip`         | 1.0.31.255                   | TEXT      | Ending IP address of an IP address range          |
| `country`        | JP                           | TEXT      | ISO 3166 country code of the location             |
| `country_name`   | Japan                        | TEXT      | Name of the country                               |
| `continent`      | AS                           | TEXT      | Continent code of the country                     |
| `continent_name` | Asia                         | TEXT      | Name of the continent                             |
| `asn`            | AS2519                       | TEXT      | Autonomous System Number                          |
| `as_name`        | ARTERIA Networks Corporation | TEXT      | Name of the AS (Autonomous System) organization   |
| `as_domain`      | arteria-net.com              | TEXT      | Official domain or website of the AS organization |

Documentation: https://ipinfo.io/developers/ip-to-country-asn-database

Samples are available here: https://github.com/ipinfo/sample-database/tree/main/IP%20to%20Country%20ASN

The database can be downloaded simply by accessing the [storage URI with an access token](https://ipinfo.io/developers/database-filename-reference).

```bash
curl -L https://ipinfo.io/data/free/country_asn.mmdb?token=<YOUR_TOKEN> -o country_asn.mmdb
```

Use the [Python MMDB library](https://pypi.org/project/maxminddb/) to read the MMDB file.

Let me know what you think. Thanks!

**Comment by rennf93** [comment]
Hi Abdullah!

Thank you very much for the suggestion. As a matter of fact, I was searching for some alternatives to my implementation.

I've now added support and integration with IPInfo's capabilities and plan to expand on these on the future.

As for now, I've released a new version that reflect the changes we've discussed; version [0.3.2](https://github.com/rennf93/fastapi-guard/releases/tag/0.3.2)
